app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/SubscriptionRecoveryController.scala

```
@@ -190,40 +219,97 @@ class SubscriptionRecoveryController @Inject() (
+    journey: Journey.Value,
+    subscriptionInformation: SubscriptionInformation
+  )(implicit hc: HeaderCarrier, messages: Messages): Future[Unit] =
+    handleSubscriptionService
+      .handleSubscription(
+        subscriptionInformation.formBundleId,
+        RecipientDetails(
+          service,
+          journey,
+          subscriptionInformation.email,
+          subscriptionInformation.recipientFullName,
+          Some(subscriptionInformation.name),
+          Some(subscriptionInformation.processedDate)
+        ),
+        TaxPayerId(subscriptionInformation.safeId.id),
+        Some(subscriptionInformation.eori),
+        subscriptionInformation.emailVerificationTimestamp,
+        subscriptionInformation.safeId
+      )
+
+  private def issue(service: Service, journey: Journey.Value, subscriptionInformation: SubscriptionInformation)(implicit
+    hc: HeaderCarrier
+  ): Future[Int] =
+    if (journey == Journey.Subscribe)
+      taxEnrolmentService.issuerCall(
+        subscriptionInformation.formBundleId,
+        subscriptionInformation.eori,
+        subscriptionInformation.dateOfEstablishment,
+        service
+      )
+    else
+      Future.successful(NO_CONTENT)
+
+  private def getDateOfBirthOrDateOfEstablishment(
+    response: SubscriptionDisplayResponse,
+    dateOfEstablishmentCaptured: Option[LocalDate],
+    dateOfBirthCaptured: Option[LocalDate]
+  )(implicit request: Request[AnyContent], headerCarrier: HeaderCarrier): Option[LocalDate] = {
+    val isIndividualOrSoleTrader = requestSessionData.isIndividualOrSoleTrader
+    val dateOfEstablishment      = response.responseDetail.dateOfEstablishment // Date we hold
+    (isIndividualOrSoleTrader, dateOfEstablishment, dateOfEstablishmentCaptured, dateOfBirthCaptured) match {
+      case (_, Some(date), _, _)     => Some(date)
+      case (false, _, Some(date), _) => Some(date)
+      case (true, _, _, Some(date))  => Some(date)
+      case _                         => None
```

I know that this is the copy of the CDS approach, but I'm wondering if we should throw an exception here. This None shouldn't happen and if happen something is wrong, because we shouldn't do an issuer call without this date

---

That's a good point.  I'll change it.  Should never be None.